MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm

```diff
@@ -266,12 +269,26 @@
 	if (isUsingDescriptorSetMetalArgumentBuffers() && isUsingMetalArgumentBuffer()) {
 		@autoreleasepool {
 			NSMutableArray<MTLArgumentDescriptor*>* args = [NSMutableArray arrayWithCapacity: _bindings.size()];
-			for (auto& dslBind : _bindings) { dslBind.addMTLArgumentDescriptors(args); }
+			for (auto& dslBind : _bindings) { dslBind.addMTLArgumentDescriptors(args, dslBind.getDescriptorCount()); }
 			_mtlArgumentEncoder.init(args.count ? [getMTLDevice() newArgumentEncoderWithArguments: args] : nil);
 		}
 	}
 }
 
+size_t MVKDescriptorSetLayout::getEncodedArgumentBufferLength(uint32_t variableDescriptorCount) {
+	if (_hasVariableDescriptorCount) {
```

Comment:
Since only the last descriptor binding in a descriptor set layout can be variable, instead of tracking `_hasVariableDescriptorCount`, querying the last binding (if at least one exists) by calling `hasVariableDescriptorCount()` is simple. 

With that idea, this could also test that `variableDescriptorCount` is less than `getDescriptorCount()` of that same binding, to determine if calculating with a temporary argument encoder is necessary.

Reply:
Good points, done.